lib/beautiful/colorful.lua

```diff
@@ -0,0 +1,1395 @@
+            self.alpha)
+end
+
+--- Returns the left and right triadic colors (120deg shift from base color)
+-- @return A new color object representing a +120 degree hue shift of the color, or nil if invalid
+-- @return A new color object representing a -120 degree hue shift of the color, or nil if invalid
+-- @usage local left, right = c:triadic()
+function color:triadic()
+    local left = self.hue + 120
+    local right = self.hue - 120
+    return color.from.hsla(
+            left,
+            self.saturation,
+            self.lightness,
+            self.alpha),
+        color.from.hsla(
+            right,
+            self.saturation,
+            self.lightness,
+            self.alpha)
+end
+
+--- Returns three colors making up the tetradic colors (base complement, base
+--+ 60deg, base + 60deg complement)
+-- @return A new color object representing the complementary color, or nil if
+-- invalid
+-- @return A new color object representing a +60 degree shift of the color, or
+-- nil if invalid
+-- @return A new color object representing the complement of the previous
+-- returned color, or nil if invalid
+-- @usage local complement, analogous, analogous_complement = c:tetradic()
+function color:tetradic()
+    local complement = self:complement()
+    local base_left = color.from.hsla(self.hue + 60, self.saturation, self.lightness, self.alpha)
+    return complement, base_left, base_left:complement()
+end
+
+-- @section end
+
+-- }}} Color harmony functions
+
+-- }}} Color transformation functions
+
+-- {{{ Color output functions
+
+--- Color object output functions
+-- @section color_output
+
+-- {{{ String output
+
+--- Return the string representation of the color in #RRGGBBAA format
+-- @treturn string A string represention of the color in #RRGGBBAA format
+-- @usage local hexa_str = c.as:hexa()
+function color.as:hexa()
+    local r = round(self.red   * 255)
+    local g = round(self.green * 255)
+    local b = round(self.blue  * 255)
+    local a = round(self.alpha * 255)
+    return string.format("#%02x%02x%02x%02x", r, g, b, a)
+end
+
+--- Return the string representation of the color in #RRGGBB format
+-- @treturn string A string represention of the color in #RRGGBB format
+-- @usage local hex_str = c.as:hex()
+function color.as:hex()
+    local r = round(self.red   * 255)
+    local g = round(self.green * 255)
+    local b = round(self.blue  * 255)
+    return string.format("#%02x%02x%02x", r, g, b)
+end
+
+-- }}} String output
+
+-- {{{ Decimal output
+
+--- Return the decimal representation of the color including the alpha channel
+-- @treturn number The decimal represention of the color including the alpha
+-- channel
+-- @usage local deca_color = c.as:deca()
+function color.as:deca()
+    return math.floor(
+          lshift(round(self.red   * 0xFF), 24)
+        + lshift(round(self.green * 0xFF), 16)
+        + lshift(round(self.blue  * 0xFF), 8)
+        +        round(self.alpha * 0xFF)
+    )
```

Comment:
This is also one of the functions which could have problems with 32 bit signed integers... I'm not sure.